.github/workflows/CI.yml

```diff
@@ -274,3 +274,32 @@ jobs:
         run: |
           SET CHERE_INVOKING=1
           C:\cygwin64\bin\bash -l -c "bash dev/ci.sh"
+
+  slack-notification-on-failure:
+    name: Send slack notification on CI failure
+    needs:
+      - test-unix
+      - test-cygwin
+    if: ${{ failure() && github.event_name == 'push' }}
```

Comment:
Although I don't think we want `failure()`, because we want to also be notified when a previously-failing branch is now passing. So I *think* this should probably be `always()`?

Reply:
Yes, totally!